docs/DeveloperGuide.md

```diff
@@ -2,37 +2,189 @@
+- **Desire for Simplicity**: They prefer tools that are easy to use, automate repetitive tasks, and function offline,
+  allowing them to streamline administrative work without unnecessary complexity.
+
+The target user values efficiency, reliability, and simplicity, seeking a solution that reduces administrative workload
+and enables them to focus more on the core aspects of teaching.
+
+### Value Proposition
+
+**TutorLink** solves the problem of administrative overload by automating routine tasks such as managing assignments,
+and monitoring student performance. Professors often struggle with time-consuming admin work that takes away from their
+primary focus: teaching and preparing lessons.
+
+By offering an offline, lightweight solution that simplifies these processes, TutorLink helps professors:
+
+- Save time by automating tedious administrative tasks.
+- Access important information quickly, without needing an internet connection.
+- Focus on teaching and lesson development instead of being bogged down by admin work.
+
+In contrast to bloated systems, TutorLink is designed to be fast, simple, and effective—freeing up valuable time and
+enhancing teaching efficiency.
 
 ## User Stories
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| Version | As a ... | I want to ...             | So that I can ...                                           |
+|---------|----------|---------------------------|-------------------------------------------------------------|
+| v1.0    | new user | see usage instructions    | refer to them when I forget how to use the application      |
+| v2.0    | user     | find a to-do item by name | locate a to-do without having to go through the entire list |
 
 ## Non-Functional Requirements
 
 {Give non-functional requirements}
 
+1. Performance: The system should respond to any command within a few seconds
+2. Reliability: The system should not lose any user data even if it crashes unexpectedly
+3. Usability: A new user should be able to use basic features without confusion
+4. Compatibility: The system should work on any mainstream OS (Windows, macOS, Linux) that has Java 11 or above
+   installed
+5. Maintainability: The system should be able to export all data in a human-readable format for backup purposes
+
 ## Glossary
 
 * *glossary item* - Definition
 
 ## Instructions for manual testing
 
 {Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+
+## Implementation
+
+### Command execution sequence: 
+
+All commands follow the sequence as described in the diagram below: 
+
+![ArchitectureSequenceGrouped.png](diagrams%2FArchitectureSequenceGrouped.png)
+
```

Comment:
![image](https://github.com/user-attachments/assets/50959538-3448-4225-b285-b2310f3dd854)

Could this constructor be omitted from this diagram, since it is not necessary to demonstrate command flow sequence?